[datadog_synthetics_test] Add `click_with_javascript` for browser tests (#3013)
DataDog/terraform-provider-datadog: docs/resources/synthetics_test.md

- `attribute` (String) Name of the attribute to use for an "assert attribute" step.
- `check` (String) Check type to use for an assertion step. Valid values are `equals`, `notEquals`, `contains`, `notContains`, `startsWith`, `notStartsWith`, `greater`, `lower`, `greaterEquals`, `lowerEquals`, `matchRegex`, `between`, `isEmpty`, `notIsEmpty`.
- `click_type` (String) Type of click to use for a "click" step.

> - `click_with_javascript` (Boolean) Whether to use `element.click()` for a `click` step. This offers a more reliable way to interact with elements but does not emulate a real user interaction.

Reviewer: ```suggestion
- `click_with_javascript` (Boolean) When true, uses `element.click()` for the `click` step. This is a more reliable way to interact with elements but does not emulate a real user interaction.
```

Author: I prefer to keep "for a ... step" because that's the pattern we use for other properties. And I feel like the sentence sounds better without a comma when I keep "for a ... step", so I'd rather keep "whether" since it's common for booleans. I'll keep the "this is" instead of "this offers" though 👍
